src/components/UI/layouts/Container/Container.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import theme from '<variables>/mainTheme';
+
+
+/**
+ * @description - Container Component
+ *
+ * @prop {node} children - children
+ * @prop {string} margin - margin
+ * @prop {string} padding - padding
+ * @prop {string} display - display
+ * @prop {string} backgroundColor - background color
+ * @prop {string} borderColor - border color
+ * @prop {bool} hasBorderRadius - add border radius
+ * @prop {string} boxShadow - boxShadow
+ *
+ * @return {component} Container
+ */
+const Container = ({
+  children,
+  margin,
+  padding,
+  display,
+  backgroundColor,
+  borderColor,
+  hasBorderRadius,
+  boxShadow,
+}) => (
+  <Container.Container
+    margin={margin}
+    padding={padding}
+    display={display}
+    backgroundColor={backgroundColor}
+    borderColor={borderColor}
+    hasBorderRadius={hasBorderRadius}
+    boxShadow={boxShadow}
+  >
+    {children}
+  </Container.Container>
+);
+
+Container.Container = styled.div`
+  ${({
+    margin,
+    padding,
+    display,
+    hasBorderRadius,
+    boxShadow,
+    backgroundColor,
+    borderColor,
+    theme: {
+      spacing, backgroundColors, borderColors, boxShadows,
+    },
+  }) => `
+    display: ${display};
+    background-color: ${backgroundColors[backgroundColor]};
+    margin: ${spacing[margin] || '0'};
+    padding: ${spacing[padding]};
+    border-radius: ${hasBorderRadius ? '15px' : '0'};
+    box-shadow: ${boxShadow
+    ? boxShadows[boxShadow] : '0'};
+    border: 1px solid ${borderColors[borderColor] || 'transparent'};
+    overflow: hidden;
+    box-sizing: border-box;
+  `}
+`;
+
+Container.defaultProps = {
+  display: 'flex',
```

Comment:
I think this should be renamed to FlexContainer so we can limit the number of things this container is doing. 
So we can have link Container, Flex Container etc.